onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc

```diff
@@ -53,9 +58,16 @@ NnapiExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_view
     }
   }
 
-  nnapi::ModelBuilder builder(graph_view);
+  int32_t android_sdk_ver;
+#ifdef __ANDROID__
+  const auto* _nnapi = NnApiImplementation();
+  android_sdk_ver = _nnapi->android_sdk_version;
+#else
+  android_sdk_ver = ORT_NNAPI_HOST_API_LEVEL;
+#endif
+
```

Comment:
I'm wondering whether we should just honor an explicitly provided level, and only fallback to getting it from the sdk if that is not provided.

In theory you could build on an android device but want to run on a different android device with a later sdk version. Maybe that's unlikely. Slightly easier to explain the behavior though (always use provided value, if not provided use current device sdk level).

Reply:
These are runtime value on Android, so we only specify the host api level on non-android system since there is no way to get this from the system